### ISSUE TITLE
man-page suggestions

### ISSUE THREAD

**Issue by EHWV** [freebsd/poudriere#945]
## Prerequisites
- [x] Have you checked for an existing issue describing the issue?

Hi there,

Thank you for developing poudriere!

Here are my suggestions for the poudriere man-pages, as copied from my message on the FreeBSD forum: [make.conf for poudriere: confusion](https://forums.freebsd.org/threads/make-conf-for-poudriere-confusion.82848/post-541375)

Because it doesn't seem possible to produce colored text here, please refer to [my message](https://forums.freebsd.org/threads/make-conf-for-poudriere-confusion.82848/post-541375) at the FreeBSD forum if you wish to see the colored text.

I realise that my remark about the naming convention of the command **options** (as used in [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports)), more specifically replacing it with another term instead of that command (for example **useopts**) probably has backwards compatibilty issues.

Kind regards,
Eric

<hr ->

![Poudriere-man-8](https://user-images.githubusercontent.com/94120901/141309985-6b02190a-d6f1-4d1b-8155-935f24a0253d.png)

![Poudriere-OPTIONS-man-8](https://user-images.githubusercontent.com/94120901/141310082-9a7ee863-d0dc-4cad-8de6-31d3949486ec.png)

At [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports) there is a part that is mentioned in the section SYNOPSIS: [_poudriere-options_] but, it is not described in that man-page (see ?-1, in the image above of [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports)).

Also, you have to realise that at the SYNOPSIS section of [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports) the string **options** is _not_ a placeholder but an actual string that has to be typed in literally. This is because [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports) is an incarnation of the general [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports) format for one specific command, namely the **options** command as specified at [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports) in its SYNOPSIS section by the _placeholder_ **command**. Compare that with [pkg(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-release-ports), there the placeholder in the SYNOPSIS section is designated in a much cleaner way by:
\<command>

IMO, it can also be unclear that in the NAME section of [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports) the following string is written as:
**poudriere-options** This is _not_ the actual command that is to be typed in. Compare for example:

1. [man(1)](https://www.freebsd.org/cgi/man.cgi?query=man&sektion=1&manpath=freebsd-release-ports)
2. [ls(1)](https://www.freebsd.org/cgi/man.cgi?query=ls&sektion=1&manpath=freebsd-release-ports)
3. [pkg(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-release-ports)

Because of the (complex) syntax of [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports), its man-pages are split into several man-pages and you have to use a unique name for a man-page; however, this gives rise to ?-2 ambiguity in the image of [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports) above. [pkg(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-release-ports) faces similar difficulties and is also split up. That might have something to do with the naming conventions or limitations of man-pages.

Firstly, however, note that with [pkg(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-release-ports) there is a subtle but _significant difference_.  [pkg(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-release-ports) is also split up into seperate man-pages, for example for the incarnation [pkg‑info(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg-info&apropos=0&sektion=8&manpath=FreeBSD+13.0-RELEASE+and+Ports&arch=default&format=html) (as mentioned in the SEE ALSO section of [pkg(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-release-ports)). The man-page of [pkg‑info(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg-info&apropos=0&sektion=8&manpath=FreeBSD+13.0-RELEASE+and+Ports&arch=default&format=html) is tagged by the string "pkg-info" but, at the NAME section of [pkg‑info(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg-info&apropos=0&sektion=8&manpath=FreeBSD+13.0-RELEASE+and+Ports&arch=default&format=html) it correctly uses the string **pkg info** (note: no dash between the two words), which string (=pkg info) is also used in that exact same way in all other sections.

Secondly, compare with **pkg-static** at the NAME section of [pkg(8)](https://www.freebsd.org/cgi/man.cgi?query=pkg&sektion=8&manpath=freebsd-release-ports) as an alternative name. **pkg-static** functions as an exact _synonym_ of **pkg**. Here the dash between the two words is mandatory.

At the SYNOPSIS section of [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports) there is a part _poudriere-options_ These options are _not_ described by a similar name in subsequent sections. They appear to be described in the section GLOBAL OPTIONS (see ?-3, in the image of [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports) above). If this is the case, why are are such different terms used to indicate the same item?

If I understand Jose's description correctly, at [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports) in the section SYNOPSIS the string [_poudriere-options_] (note: the presence of the two "[" & "]") is not described in [poudriere-options(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere-options&sektion=8&manpath=freebsd-release-ports) but _is_ described in the section GLOBAL OPTIONS of [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports). If this is indeed the case, I do not find that self-evident. The name change into "GLOBAL OPTIONS" does not help. I have not seen any other man-pages where you have to cross reference to find a description of a certain item for which there is not an expliciet cross-reference present. I may be wrong; if so, I'd appreciate examples where that is the case.

It seems that the use of the word _argument(s)_ of a command is not in line with the usage elsewhere in man-pages. With the command poudriere of [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports) the first (optional) argument is [_poudriere-options_]. The second argument is **command**. In the COMMANDS section of [poudriere(8)](https://www.freebsd.org/cgi/man.cgi?query=poudriere&sektion=8&manpath=freebsd-release-ports) this is described as:
> The first argument to poudriere must be a command from the following list:

This could be reworded as:
\<command> must be one of the following list:

This also begs the question if, instead of the command/argument **options**, something like **useopts** or **portoptions** would not have been a more apropriate term, not clashing with the terminology _option(s)_ used pervasively in man-pages.

<hr ->

Perhaps this (differences highlighted) is clearer:

![ALT-poudriere](https://user-images.githubusercontent.com/94120901/141310141-11abc4c2-f961-4c65-8aa1-9004e9b2428b.png)

![ALT-options](https://user-images.githubusercontent.com/94120901/141310186-513f0878-b231-4784-ba9b-77dd3867137d.png)

**Comment by bdrewery** [comment]
I like where you are going with this. The current naming scheme is a disaster of inconsistency and confusing. Maybe "flags" instead of "options".

Do you know the mdoc manpage syntax? It's easier to learn than you might think. `man mdoc` for reference. Then you can modify a manpage and do `man path/to/file.1` to render it. The only weird gotcha that I can't explain is that any sentence ending in a period (`.`) should be followed by a newline.

Otherwise someone might find time to work on this, but I encourage you to give it a try.

**Comment by grahamperrin** [comment]
(Moved from <https://github.com/freebsd/poudriere/discussions/944#discussioncomment-1627685>)

Thanks! The topic in FreeBSD Forums is quite sprawling, so I'll offer this cherry-pick: 

> That single red arrow beneath the heading **POUDRIERE(8)** was enough for an immediate, at-a-glance realisation.

**tl;dr** where <https://github.com/freebsd/poudriere/wiki/poudriere-options.8#SYNOPSIS> currently shows: 

> [_poudriere-options_]

– I suspect that the phrase: 

> [_global options_]

– will be less ambiguous; immediately understandable.

**Comment by grahamperrin** [comment]
Also <https://github.com/freebsd/poudriere/wiki/poudriere-testport.8#SYNOPSIS> (but not <https://github.com/freebsd/poudriere/wiki/poudriere-testport.8-devel#SYNOPSIS>)
